pkg/controller/kubefedcluster/controller.go

```diff
@@ -219,21 +238,35 @@ func (cc *ClusterController) updateIndividualClusterStatus(cluster *fedv1b1.Kube
 	storedData *ClusterData, wg *sync.WaitGroup) {
 	clusterClient := storedData.clusterKubeClient
 
-	currentClusterStatus := clusterClient.GetClusterHealthStatus()
+	cc.RecordEvent(cluster, "UpdatingStatus", "Updating the status of the cluster")
+
+	currentClusterStatus, err := clusterClient.GetClusterHealthStatus()
+	if err != nil {
+		cc.RecordError(cluster, "RetrievingClusterHealthFailed", errors.Wrap(err, "Failed to retrieve health of the cluster"))
+	}
+
 	currentClusterStatus = thresholdAdjustedClusterStatus(currentClusterStatus, storedData, cc.clusterHealthCheckConfig)
 
 	if utilfeature.DefaultFeatureGate.Enabled(features.CrossClusterServiceDiscovery) {
-		currentClusterStatus = updateClusterZonesAndRegion(currentClusterStatus, cluster, clusterClient)
+		currentClusterStatus = cc.updateClusterZonesAndRegion(currentClusterStatus, cluster, clusterClient)
 	}
 
 	storedData.clusterStatus = currentClusterStatus
 	cluster.Status = *currentClusterStatus
 	if err := cc.client.UpdateStatus(context.TODO(), cluster); err != nil {
-		klog.Warningf("Failed to update the status of cluster %q: %v", cluster.Name, err)
+		cc.RecordError(cluster, "UpdatingStatusFailed", errors.Wrap(err, "Failed to update the status of the cluster"))
```

Comment:
I'm not sure I understand how this event would be useful to a user.  What error conditions would it indicate, and how would the event allow them to identify or fix the problem?

Reply:
When status update failed, I think the event can tell there has an issue. It tells that the status is not updated correctly. I'm not sure if `err` information helps for fix the problem but it tells the problem exists.

Reply:
Are there other kube resources for which failure to report status is recorded as an event? I can think of at least 2 reasons for which an event for failure to update status should not be logged:

 - concurrency (i.e. an update was made to the resource by another party) since this is recoverable
 - transitory connectivity problem (since this is likely recoverable as well)

Consider that in both cases the error is likely to be transitory and automatically recovered.  Logging yes, but an event?  It will just clutter the log.

Reply:
agreed. Updated